File: day2/part1.py
def isValid(number):
    line = str(number)
    if len(line) % 2 != 0:
        return True
    firstHalf = line[:len(line)//2]
    secondHalf = line[len(line)//2:]
    return not firstHalf == secondHalf


def getNumbersFromRanges(file):
    numbers = []
    ranges = file.split(",")
    for r in ranges:
        print(r)
        cleanline = r.strip()
        parts = r.split("-")
        start = int(parts[0])
        end = int(parts[1])
        print(f"Generating numbers from {start} to {end}")
        # for i in range(start, end + 1):
        #     print(i)
        #     numbers.append(i)
        numbers.extend(range(start, end + 1))
    return numbers



# Open the file in read mode and use readlines()
with open("input.txt", "r") as f:
    file = f.read()
    numbers = getNumbersFromRanges(file)
    invalidNumbers = [number for number in numbers if not isValid(number)]
    print(invalidNumbers)
    print(len(invalidNumbers))
    print(sum(invalidNumbers))

File: day2/Part1.java
import java.io.*;
import java.util.*;
import java.util.stream.*;

public class Part1 {

    static boolean isValid(int number) {
        String line = String.valueOf(number);
        if (line.length() % 2 != 0) {
            return true;
        }
        String firstHalf = line.substring(0, line.length() / 2);
        String secondHalf = line.substring(line.length() / 2);
        return !firstHalf.equals(secondHalf);
    }

    static List<Integer> getNumbersFromRanges(String file) {
        List<Integer> numbers = new ArrayList<>();
        String[] ranges = file.split(",");
        for (String r : ranges) {
            System.out.println(r);
            String cleanLine = r.trim();
            String[] parts = cleanLine.split("-");
            int start = Integer.parseInt(parts[0].trim());
            int end = Integer.parseInt(parts[1].trim());
            System.out.println("Generating numbers from " + start + " to " + end);
            for (int i = start; i <= end; i++) {
                numbers.add(i);
            }
        }
        return numbers;
    }

    public static void main(String[] args) throws IOException {
        String file;
        try (BufferedReader br = new BufferedReader(new FileReader("input.txt"))) {
            file = br.lines().collect(Collectors.joining(","));
        }

        List<Integer> numbers = getNumbersFromRanges(file);
        List<Integer> invalidNumbers = numbers.stream()
                .filter(n -> !isValid(n))
                .collect(Collectors.toList());

        System.out.println(invalidNumbers);
        System.out.println(invalidNumbers.size());
        System.out.println(invalidNumbers.stream().mapToLong(Integer::longValue).sum());
    }
}
